src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -0,0 +1,929 @@
+                                                  stream);
+
+                pa_stream_set_started_callback( stream->inStream,
+                                                PaPulseAudio_StreamStartedCb,
+                                                stream );
+            }
+            else
+            {
+                PA_DEBUG( ("Portaudio %s: Can't read audio!\n",
+                          __FUNCTION__) );
+
+                goto startstreamcb_error;
+            }
+            PaPulseAudio_UnLock( l_ptrPulseAudioHostApi->mainloop );
+        }
+        else
+        {
+            goto startstreamcb_error;
+        }
+
+    }
+
+    if( stream->outStream ||
+        stream->inStream )
+    {
+        stream->isActive = 0;
+        stream->isStopped = 1;
+
+        while( 1 )
+        {
+            if( stream->outStream != NULL )
+            {
+                PaPulseAudio_Lock( l_ptrPulseAudioHostApi->mainloop );
+                l_SState = pa_stream_get_state( stream->outStream );
+                PaPulseAudio_UnLock( l_ptrPulseAudioHostApi->mainloop );
+
+                if( PA_STREAM_READY == l_SState &&
+                    !l_iPlaybackStreamStarted )
+                {
+                    l_iPlaybackStreamStarted = 1;
+                }
+                else if( PA_STREAM_FAILED == l_SState ||
+                         PA_STREAM_TERMINATED == l_SState )
+                {
+                    goto startstreamcb_error;
+                }
+            }
+
+            if( stream->inStream != NULL )
+            {
+                PaPulseAudio_Lock( l_ptrPulseAudioHostApi->mainloop );
+                l_SState = pa_stream_get_state( stream->inStream );
+                PaPulseAudio_UnLock( l_ptrPulseAudioHostApi->mainloop );
+
+                if( PA_STREAM_READY == l_SState &&
+                    !l_iRecordStreamStarted )
+                {
+                    l_iRecordStreamStarted = 1;
+                }
+                else if( PA_STREAM_FAILED == l_SState ||
+                         PA_STREAM_TERMINATED == l_SState )
+                {
+                    goto startstreamcb_error;
+                }
+
+            }
+
+            if( !stream->inStream && !stream->outStream)
+            {
+                goto startstreamcb_error;
+            }
+
+            if( (l_iPlaybackStreamStarted && !stream->inStream) ||
+                (l_iRecordStreamStarted && !stream->outStream) ||
+                ((l_iPlaybackStreamStarted && l_iRecordStreamStarted) &&
+                (stream->inStream && stream->outStream)) )
+            {
+                stream->isActive = 1;
+                stream->isStopped = 0;
+                break;
+            }
+
+            l_iLoop ++;
+
+            if( l_iLoop >= 100)
+            {
+                PA_DEBUG( ("Portaudio %s: Can't connect streams!\n",
+                          __FUNCTION__) );
+                goto startstreamcb_error;
+            }
+
+            usleep(1000);
```

Comment:
It turns out that 1 ms * 100 is too short to be always successful.  I have changed it to 10 ms (10000) and it works without failure.